packages/braintree-utils/src/braintree.ts

```diff
@@ -883,6 +884,212 @@ export interface BraintreeConnectCardComponent {
+    root?: {
+        backgroundColorPrimary?: string;
+        errorColor?: string;
+        fontFamily?: string;
+    };
+    input?: {
+        borderRadius?: string;
+        borderColor?: string;
+        focusBorderColor?: string;
+    };
+    toggle?: {
+        colorPrimary?: string;
+        colorSecondary?: string;
+    };
+    text?: {
+        body?: {
+            color?: string;
+            fontSize?: string;
+        };
+        caption?: {
+            color?: string;
+            fontSize?: string;
+        };
+    };
+    branding?: string; // 'light' | 'dark'
+}
+
+export enum BraintreeFastlaneAuthenticationState {
+    SUCCEEDED = 'succeeded',
+    FAILED = 'failed',
+    CANCELED = 'cancelled',
+    UNRECOGNIZED = 'unrecognized',
+}
+
+export interface BraintreeFastlaneAuthenticationCustomerResult {
+    authenticationState: BraintreeFastlaneAuthenticationState;
+    profileData: BraintreeFastlaneProfileData;
+}
+
+export interface BraintreeFastlaneProfileData {
+    fastlaneCustomerAuthAssertionToken: string;
+    fastlaneCustomerId: string;
+    shippingAddress: BraintreeFastlaneAddress;
+    card: BraintreeFastlaneVaultedInstrument;
+    name: {
+        given_name: string;
+        surname: string;
+    };
+}
+
+export interface BraintreeFastlaneAddress {
+    id?: string;
+    firstName?: string;
+    lastName?: string;
+    company?: string;
+    streetAddress: string;
+    extendedAddress?: string;
+    locality: string;
+    region: string;
+    postalCode: string;
+    countryCodeNumeric?: number;
+    countryCodeAlpha2: string;
+    countryCodeAlpha3?: string;
+}
+
+export interface BraintreeFastlaneCardPaymentSource {
+    brand: string;
+    expiry: string;
+    lastDigits: string;
+    name?: string;
+    billingAddress: BraintreeFastlaneAddress;
+}
+
+export interface BraintreeFastlanePaymentSource {
+    card: BraintreeFastlaneCardPaymentSource;
+}
+
+export interface BraintreeFastlaneVaultedInstrument {
+    id: string; // This is the nonce / token
+    paymentSource: BraintreeFastlanePaymentSource;
+}
+
+export interface BraintreeFastlaneCardComponentOptions {
+    fields: BraintreeFastlaneCardComponentFields;
+}
+
+export interface BraintreeFastlaneCardComponentFields {
+    [key: string]: BraintreeFastlaneCardComponentField;
+}
+export interface BraintreeFastlaneCardComponentField {
+    placeholder?: string;
+    prefill?: string;
+}
+
+export interface BraintreeFastlaneTokenizeResult {
+    nonce: string;
+    details: BraintreeFastlaneTokenizeDetails;
+    description: string;
+    type: string;
+}
+
+export interface BraintreeFastlaneTokenizeDetails {
+    bin: string;
+    cardType: string;
+    expirationMoth: string;
+    expirationYear: string;
+    cardholderName: string;
+    lastFour: string;
+    lastTwo: string;
+}
+
+export interface BraintreeFastlaneTokenizeOptions {
+    billingAddress?: BraintreeFastlaneAddress;
+    shippingAddress?: BraintreeFastlaneAddress;
+}
```

Comment:
Can you check the integration doc to be sure that there is nothing missing for tokenize method options?

Reply:
Sure, i've checked. Here is interface specified in integration docs 
`interface TokenizeOptions {
 billingAddress: Address;
 shippingAddress: Address; (optional)
}`

Reply:
This is quite strange, because PayPal team told us about this change on one of the meeting. Can you double check this with PayPal team in our external chat?

Reply:
It's ok to keep it like this for now, but we should change it in the future